src/components/Exercises.tsx

```diff
@@ -1,32 +1,31 @@
-import axios from "axios"
-import { useEffect, useState } from "react"
+import { useContext, useEffect, useState } from "react"
+import AuthContext from "../context/AuthProvider";
 import { Exercise } from "../models/exercise"
+import { getExercisesByUserId } from "../services/apiService";
+import { decodeToken } from "../services/tokenService";
 
 interface Props {
 
 }
 
 export const Exercises = (props: Props) => {
+    const { auth } = useContext(AuthContext);
     const [exercises, setExercises] = useState<Exercise[]>([])
 
-    const getUsers = async () => {
-        const res = await axios.get(`https://rm-tracker-357607.uc.r.appspot.com/exercises`)
-
-        setExercises(res.data)
-    }
-
-    console.log("exercises", exercises)
-
     useEffect(() => {
-        getUsers()
-    }, [])
+        async function fetchData() {
+            const { id } = decodeToken(auth.token)
+            const exercises = await getExercisesByUserId(id)
+            setExercises(exercises)
+        }
```

Comment:
Define `fetchData` ouside of `useEffect` so that it isn't created every time `useEffect` is run. Also, the name`fetchData` is a bit misleading, as it fetches and sets. Maybe `fetchAndSetExercises` would be clearer.